### PR TITLE
Add Batch Create Endpoint

### PR DESCRIPTION
**Pull request by joelip**
Mentioned this in slack, but I'm having trouble with hitting this endpoint in VCR...

If you want to wait until I can get it working that's fine. I'll try and fix it tomorrow.

Thanks man!

### CODE REVIEW

File: lib/client/checkpoints.rb [hunk 1]
```diff
@@ -44,4 +44,39 @@ def update_checkpoint(id, options={})
...
+  #         description: 'Message from Tamriel',
+  #         body: 'Listen to the Nine',
+  #         position: 1
+  #       },
+  #       ...
+  #     ]
+  #   )
+
+  def batch_create_checkpoints(roadmap_section_id=nil, checkpoints=[])
+    url = "#{@api_base_path}/checkpoints/batch_create"
```
Comment (getaaron) [inline lib/client/checkpoints.rb]: I would use single quotes here and on line 79, because they're more stylish

File: lib/client/checkpoints.rb [hunk 2]
```diff
@@ -44,4 +44,39 @@ def update_checkpoint(id, options={})
       convert_response(response, "checkpoint")
```
Comment (getaaron) [inline lib/client/checkpoints.rb]: also here

File: lib/client/checkpoints.rb (hunk 1 above)
Comment (bmneely) [inline lib/client/checkpoints.rb]: You can't use single quotes with Ruby string interpolation.

File: lib/client/checkpoints.rb (hunk 2 above)
Comment (bmneely) [inline lib/client/checkpoints.rb]: I agree here thought

File: lib/client/checkpoints.rb (hunk 1 above)
Comment (getaaron) [inline lib/client/checkpoints.rb]: Hmm how about

```ruby
url = api_base_path + '/' + 'checkpoints' + '/' + "batch_create"
```

Comment (bmneely) [inline lib/client/checkpoints.rb]: No, that's bad and you are bad for suggesting it.

Comment (mikesjewett) [inline lib/client/checkpoints.rb]: I agree with both of you.

File: lib/client/checkpoints.rb (hunk 2 above)
Comment (mikesjewett) [inline lib/client/checkpoints.rb]: I disagree with both of you.

Comment (bmneely) [inline lib/client/checkpoints.rb]: I do think you should rename the method to be `setup_response`, right @mikesjewett?

Comment (mikesjewett) [inline lib/client/checkpoints.rb]: yes and no

File: lib/client/checkpoints.rb (hunk 1 above)
Comment (joelip) [inline lib/client/checkpoints.rb]: lol

File: lib/client/checkpoints.rb [hunk 3]
```diff
@@ -44,4 +44,39 @@ def update_checkpoint(id, options={})
...
+  # @param roadmap_section_id [Integer] id for roadmap section the checkpoints belong to.
+  # @return [Array] Array of the created checkpoints.
+  # @example Create multiple checkpoints
+  #   Spartacus#batch_create_checkpoints(
+  #     163,
+  #     [
+  #       {
+  #         ids: [],
+  #         name: 'Sample checkpoint 1',
+  #         description: 'Message from Tamriel',
```
Comment (joelip) [inline lib/client/checkpoints.rb]: note to self, description is not a valid param...it's summary

**Pull request merged**